Allow filtering of data when some keyed columns contain null

PhilipRieck opened the pull request:
If you have a collection where some of the objects have a property that resolves to `null`, and you set the filter-keys to contain that property, the `doFilter` method would stop with an error trying to call `toString()` on the null value.

tochoromero commented on src/au-table.js:
```diff
@@ -101,14 +101,17 @@ export class AureliaTableCustomAttribute {
-                if (value.toLowerCase().indexOf(this.filterText.toLowerCase()) > -1) {
-                    filteredData.push(next);
-                    break;
+                if( next[key]){
```
If the property in question has a value of 0, it will evaluate to false.

We need to check if next[key] == null || next[key] == undefined

PhilipRieck replied:
great point, although `next[key] != null` also checks for undefined due to coercion.  I'll fix this in the pull request.

tochoromero commented:
Thank you for sending the pull request, I added a small note.
Also, for some reason your build is adding one extra space in one of the pagination files, I guess you can omit those from the commit.

PhilipRieck commented:
I reverted the whitespace change on the build (sorry!).   Great catch on `0` values being ommited, I've fixed that as well.
Great simple component by the way - love it.

tochoromero commented:
I have merged your pull request. Thank you for the fix :)

I'm glad you like the table, this is my very first open source contribution, please don't hesitate to send any comments or suggestions.
